src/eterna/mode/GameMode.ts

```diff
@@ -283,6 +284,33 @@ export default abstract class GameMode extends AppMode {
     protected onSetPip(_pipMode: boolean): void {
     }
 
+    protected addPose3D(structureFile: string | File | Blob) {
+        const ublk = this.getCurrentUndoBlock(0);
+        Assert.assertIsDefined(ublk, "Can't create 3D view - undo state not available");
+        if (this._pose3D) this.removeObject(this._pose3D);
+        this._pose3D = new Pose3D(structureFile, ublk.sequence, ublk.targetPairs, this._poses[0].customNumbering);
+        this.addObject(this._pose3D, this.dialogLayer);
+        this.regs?.add(this._pose3D.baseHovered.connect((closestIndex) => {
```

Comment:
this is a longer term question, honestly, but it's a shame that regs is by definition regs-or-null just because (as far as I can remember) it's not defined at initialization, but to have a functional object afterwards it has to get initialized. it requires all these asserts or conditional dots, and it's silly because if regs is null the whole thing's coming down.

Reply:
This is a flashbang idiosyncrasy - regs is nulled out when the mode is destroyed. We certainly should not be initializing it, but I'm not sure that the assert is particularly necessary since we should never be in that state, and worst case if we are, it just means that the invalid action of registering listeners on a nonexistant mode just does nothing. Admittedly, this would mean we have connected up the listeners in a way where we may have dangling listeners.

Reply:
yeah, longer term we should have a refactor that doesn't guard against methods being called on objects that don't exist. (mechanism being, like, not complicating types entirely so we can null members upon destruction, which... i don't THINK is necessary?)